monitoring/nethealth.go

```diff
@@ -396,6 +408,35 @@ func getPeerName(labels []*dto.LabelPair) (peer string, err error) {
 	return "", trace.NotFound("unable to find %s label", peerLabel)
 }
 
+// filterNetData filters the networkData. Nethealth may retain metrics for nodes
+// that have previously left the cluster. This data is now irrelevant and should
+// not be further processed.
+func (c *nethealthChecker) filterNetData(netData map[string]networkData) (filtered map[string]networkData, err error) {
+	members, err := c.SerfClient.Members()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return filterBySerf(netData, members)
+}
+
+// filterBySerf removes netData for nodes that are not part of the serf cluster.
+func filterBySerf(netData map[string]networkData, members []serf.Member) (filtered map[string]networkData, err error) {
+	filtered = make(map[string]networkData)
+	for _, member := range members {
+		data, exists := netData[member.Addr.String()]
+		if !exists {
+			log.Warnf("Missing nethealth data for node %s", member.Addr.String())
+			continue
+		}
+		if agent.MemberStatus(member.Status) != agent.MemberAlive {
```

Comment:
So how does this work now if you block udp traffic? Presumably the serf member will become "failed" and the checker won't show anything? Maybe we should be skipping over "left" members only here?

Reply:
Looks like serf doesn't communicate through the overlay network, so they won't become `failed` in this situation. But yup, if tcp is blocked as well this would prevent the check from validating.